repository/src/main/java/com/epam/esm/repository/impl/TagRepositoryImpl.java

```diff
@@ -1,79 +1,61 @@
 package com.epam.esm.repository.impl;
 
 import com.epam.esm.entity.Tag;
-import com.epam.esm.repository.mapper.TagMapper;
 import com.epam.esm.repository.repositoryinterfaces.TagRepository;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
-import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
-import org.springframework.jdbc.core.namedparam.SqlParameterSource;
-import org.springframework.jdbc.support.GeneratedKeyHolder;
-import org.springframework.jdbc.support.KeyHolder;
 import org.springframework.stereotype.Repository;
 
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
 import java.util.List;
-import java.util.Objects;
 import java.util.Optional;
 
-import static com.epam.esm.repository.Parameters.*;
 import static com.epam.esm.repository.builder.TagQueries.*;
 
 @Repository
 public class TagRepositoryImpl implements TagRepository {
 
-    private final TagMapper rowMapper;
-    private final NamedParameterJdbcTemplate namedParameterJdbcTemplate;
+    @PersistenceContext
+    private EntityManager entityManager;
 
-    @Autowired
-    public TagRepositoryImpl(TagMapper rowMapper, NamedParameterJdbcTemplate jdbcTemplate) {
-        this.rowMapper = rowMapper;
-        this.namedParameterJdbcTemplate = jdbcTemplate;
+    public TagRepositoryImpl(EntityManager entityManager) {
+        this.entityManager = entityManager;
     }
 
     @Override
     public List<Tag> findAll() {
-        return namedParameterJdbcTemplate.query(SELECT_ALL_TAGS, rowMapper);
+        return entityManager.createQuery(SELECT_ALL_TAGS, Tag.class).getResultList();
     }
 
     @Override
     public Optional<Tag> findById(Long id) {
-        SqlParameterSource sqlParameterSource = new MapSqlParameterSource().addValue(ID_PARAMETER, id);
-
-        List<Tag> tagList = namedParameterJdbcTemplate.query(SELECT_TAG_BY_ID, sqlParameterSource, rowMapper);
-
-        return tagList.stream().findFirst();
+        return Optional.of(entityManager.find(Tag.class, id));

```

Comment:
I think the ofNullable() method should also be applied here

Reply:
fixed